TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -771,6 +771,32 @@ public class DataDescription: DataDefinition, Parent<DataDescription>{
         [NotNull]
         public new DataDescriptionEntry CodeElement => (DataDescriptionEntry) base.CodeElement;
 
+        public bool IsDataDescriptionGroup
+        {
+            get
+            {
+                if (this.CodeElement?.LevelNumber?.Value < 50 &&
+                    this.TypeDefinition == null && this.Picture == null && !this.CodeElement.ConsumedTokens.Any(ct => ct.SourceText.Equals("Type", StringComparison.OrdinalIgnoreCase)))
```

Comment:
Isn't the test on ConsumedTokens redundant with the test on TypeDefinition ?

Reply:
The test on ConsumedTokens is relevant when the Type is not defined. The value of TypeDefinition will be null though the DataDef isn't a data structure.